test/unit/plugins/provisioners/ansible/provisioner_test.rb

```diff
@@ -583,6 +583,33 @@ def self.it_should_create_and_use_generated_inventory(with_ssh_user = true)
       end
     end
 
+    describe "with galaxy support" do
+
+      before do
+        config.galaxy_role_file = existing_file
+      end
+
+      it "raises an error when ansible-galaxy command fails", skip_before: true, skip_after: true do
+        config.finalize!
+        Vagrant::Util::Subprocess.stub(execute: Vagrant::Util::Subprocess::Result.new(1, "", ""))
+
+        expect {subject.provision}.to raise_error(VagrantPlugins::Ansible::Errors::AnsibleCommandFailed)
+      end
+
+      it "execute ansible-galaxy and ansible-playbook" do
+        # TODO: to be improved, but I'm currenty facing some issues, maybe only present in RSpec 2.14...
```

Comment:
@sethvargo I'd like to have a test structure like the following (with more checks, when it will work):

```ruby
        expect(Vagrant::Util::Subprocess).to receive(:execute).with { |*args|
          expect(args[0]).to eq("ansible-galaxy")
          true
        }.ordered
        expect(Vagrant::Util::Subprocess).to receive(:execute).with { |*args|
          expect(args[0]).to eq("ansible-playbook")
          true
        }.ordered
```

But this kind of sequence fails for me,

```
Failures:

  1) VagrantPlugins::Ansible::Provisioner::Host#provision with galaxy support execute ansible-galaxy and ansible-playbook
     Failure/Error: expect(args[0]).to eq("ansible-galaxy")
       
       expected: "ansible-galaxy"
            got: "ansible-playbook"
       
       (compared using ==)
     # ./test/unit/plugins/provisioners/ansible/provisioner_test.rb:617:in `block (5 levels) in <top (required)>'
     # ./plugins/provisioners/ansible/provisioner/host.rb:76:in `execute_command_from_host'
     # ./plugins/provisioners/ansible/provisioner/host.rb:124:in `execute_ansible_playbook_from_host'
     # ./plugins/provisioners/ansible/provisioner/host.rb:23:in `provision'
     # ./test/unit/plugins/provisioners/ansible/provisioner_test.rb:202:in `block (3 levels) in <top (required)>'
```
...apparently because apparently RSpec applies the `expect(args[0]).to eq("ansible-galaxy")` expectation not only to the first `:execute` method call, but to both ("ansible-galaxy" and "ansible-playbook").

I've searched a little across the RSpec docs, stackoverflow questions, etc. but so far without big results (I can find back the links if you want).

Seth, It would be awesome if you catch it for me :)